src/annotator/components/Toolbar.tsx

```diff
@@ -1,84 +1,107 @@
-import { IconButton } from '@hypothesis/frontend-shared';
+import {
+  ButtonBase,
+  AnnotateIcon,
+  CancelIcon,
+  CaretRightIcon,
+  CaretLeftIcon,
+  HideIcon,
+  NoteIcon,
+  ShowIcon,
+} from '@hypothesis/frontend-shared/lib/next';
+import type {
+  IconComponent,
+  PresentationalProps,
+} from '@hypothesis/frontend-shared/lib/types';
+import type { ButtonCommonProps } from '@hypothesis/frontend-shared/lib/components/input/ButtonBase';
 import classnames from 'classnames';
+import type { JSX, RefObject } from 'preact';
 
-/**
- * @typedef {import("@hypothesis/frontend-shared/lib/components/buttons").IconButtonProps} IconButtonProps
- *
- * @typedef {Omit<IconButtonProps, "className">} ToolbarButtonProps
- */
+// TODO: ToolbarButton should be extracted as a shared design pattern or
+// component
+type ToolbarButtonProps = PresentationalProps &
+  ButtonCommonProps &
+  Omit<JSX.HTMLAttributes<HTMLButtonElement>, 'icon' | 'size'> & {
+    icon: IconComponent;
+  };
```

Comment:
What the comment says, plus, we need to get types in order in the `frontend-shared` packge.